libs/foundry-dev-tools/src/foundry_dev_tools/resources/dataset.py

```diff
@@ -10,6 +10,9 @@
 from pathlib import Path
 from typing import TYPE_CHECKING, ClassVar, overload
 
+import polars as pl
```

Comment:
Please move imports into the if TYPE_CHECKING block as polars is an optional dependency.

Reply:
As polars will be required in ds.to_polars(), it could not be an optional dependency anymore, unfortunately. This is definitely a drawback of the PR.

Reply:
Good point @nicornk. I updated the PR to still keep polars optional.